### PR TITLE
Process relative pathnames in URLPatternInit. (Fixes #106)

### LINKED ISSUE #106: spec is missing algorithm for relative pathname patterns
The spec appears to be missing the steps necessary to process relative pathname patterns.  This is represented by this code in the c++:

https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/url_pattern/url_pattern.cc;l=149-169;drc=9e20a8d10d90b04cfee4930da49704efc28ce418

### PR DESCRIPTION
**Pull request by wanderview**
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/urlpattern/pull/107.html" title="Last updated on Aug 31, 2021, 5:30 PM UTC (a145466)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/urlpattern/107/718b9f9...a145466.html" title="Last updated on Aug 31, 2021, 5:30 PM UTC (a145466)">Diff</a>

### CODE REVIEW

**Comment by wanderview** [comment]
@domenic can you please review this pull request?  The related c++ code is:

https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/url_pattern/url_pattern.cc;l=149-169;drc=9e20a8d10d90b04cfee4930da49704efc28ce418

File: spec.bs [hunk 1]
```diff
@@ -1584,12 +1586,38 @@ To <dfn>convert a modifier to a string</dfn> given a [=part/modifier=] |modifier
   1. If |init|["{{URLPatternInit/password}}"] is not null then set |result|["{{URLPatternInit/password}}"] to the result of [=process password for init=] given |init|["{{URLPatternInit/password}}"] and |type|.
   1. If |init|["{{URLPatternInit/hostname}}"] is not null then set |result|["{{URLPatternInit/hostname}}"] to the result of [=process hostname for init=] given |init|["{{URLPatternInit/hostname}}"] and |type|.
   1. If |init|["{{URLPatternInit/port}}"] is not null then set |result|["{{URLPatternInit/port}}"] to the result of [=process port for init=] given |init|["{{URLPatternInit/port}}"], |result|["{{URLPatternInit/protocol}}"], and |type|.
-  1. If |init|["{{URLPatternInit/pathname}}"] is not null then set |result|["{{URLPatternInit/pathname}}"] to the result of [=process pathname for init=] given |init|["{{URLPatternInit/pathname}}"], |result|["{{URLPatternInit/protocol}}"], and |type|.
+  1. If |init|["{{URLPatternInit/pathname}}"] is not null:
+    1. Set |result|["{{URLPatternInit/pathname}}"] to |init|["{{URLPatternInit/pathname}}"].
+    1. If the following are all true:
+      <ol>
```
Comment (domenic) [inline spec.bs:1592]: I would probably make this a `<ul>`.

(Also, I think with enough whitespace you can make the Markdown-style syntax work, but probably not worth the effort.)

File: spec.bs [hunk 2]
```diff
@@ -1584,12 +1586,38 @@ To <dfn>convert a modifier to a string</dfn> given a [=part/modifier=] |modifier
...
+        <li>|baseURL| is not null;</li>
+        <li>|baseURL|'s [=cannot be a base URL flag=] is false; and</li>
+        <li>the result of running [=is an absolute pathname=] given |result|["{{URLPatternInit/pathname}}"] and |type| is false,
+      </ol>
+      <p>then:
+      1. Let |slash index| be the index of the last U+002F (`/`) code point found in |baseURL|'s [=url/API pathname string=] or null if there are no instances of the code point.
```
Comment (domenic) [inline spec.bs:1598]:
```suggestion
      1. Let |slash index| be the index of the last U+002F (`/`) code point found in |baseURL|'s [=url/API pathname string=], interpreted as a sequence of [=code points=], or null if there are no instances of the code point.
```

File: spec.bs [hunk 3]
```diff
@@ -1584,12 +1586,38 @@ To <dfn>convert a modifier to a string</dfn> given a [=part/modifier=] |modifier
...
+      1. If |slash index| is not null:
+        1. Let |new pathname| be the [=code point substring by indices|code point substring=] from indices 0 to |slash index| inclusive within |baseURL|'s [=url/API pathname string=].
+        1. Append |result|["{{URLPatternInit/pathname}}"] to the end of |new pathname|.
+        1. Set |result|["{{URLPatternInit/pathname}}" to |new pathname|.
```
Comment (domenic) [inline spec.bs:1602]:
```suggestion
        1. Set |result|["{{URLPatternInit/pathname}}"] to |new pathname|.
```

**Review by domenic — APPROVED** [review]

**Pull request merged**